scripts/src/models/orders.ts

```diff
@@ -123,3 +143,18 @@ export class Order extends CreationDateModel {
 		}
 	}
 }
+
+@Register
+@Initializer("origin", () => MarketplaceOrder.CLASS_ORIGIN)
+export class MarketplaceOrder extends Order {
+	public static readonly CLASS_ORIGIN = "marketplace";
+}
+
+@Register
+@Initializer("origin", () => ApplicationOrder.CLASS_ORIGIN)
+export class ApplicationOrder extends Order {
+	public static readonly CLASS_ORIGIN = "application";
+
+	@Column({ name: "wallet_address" })
+	public readonly walletAddress!: string;
```

Comment:
Not sure this will work. I’d just add wallet address to all orders and have a single class per table. This is standard ORM usage. The model should just provide data access and zero application logic.
You could implement extra logic in composition in a different file